### PR TITLE
issue#511: Add custom domain data source

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

This PR adds a new data source to retrieve the custom domain configuration.

### 📚 References

- #511 

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: internal/auth0/customdomain/data_source.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		Description: "Data source to retrieve the custom domain configuration.",
+		Schema:      dataSourceSchema(),
+	}
+}
+
+func dataSourceSchema() map[string]*schema.Schema {
+	return internalSchema.TransformResourceToDataSource(NewResource().Schema)
+}
+
+func readCustomDomainForDataSource(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/auth0/customdomain/data_source.go:27]: The meta property has access to the already configured api client, essentially it is the api client at the moment, however in the future we could create a struct that holds both the api client and the key/value mutex used throughout other resources. In this way we wouldn't use the mutex as a global property any more.

The context isn't used anywhere sadly 😢 but needs to be present within the function signature.

[the remark this answers is not in the document]
Comment (Widcket) [reply internal/auth0/customdomain/data_source.go:27]: Thanks for the context (put not intended). Maybe mark it as unused with `_` then?

[the remark this answers is not in the document]
Comment (Widcket) [reply internal/auth0/customdomain/data_source.go:27]: Nevermind, too late 😬, my bad.

[the remark this answers is not in the document]
Comment (sergiught) [reply internal/auth0/customdomain/data_source.go:27]: Argh, sorry already merged 😢 I can follow up on this, maybe tweaking our linter to check and fix such occurences.

**Pull request merged**